internal/decode/decode.go

```diff
@@ -170,7 +171,16 @@ func decodeFrom(src interface{}) Into {
 			return fmt.Errorf("failed to set up decoder: %v", err)
 		}
 
-		return decoder.Decode(src)
+		if err := decoder.Decode(src); err != nil {
+			// If we get a mapstructure error, pull the individual errors from
+			// it and use our standard multierr package.
+			if merr, ok := err.(*mapstructure.Error); ok {
+				return multierr.Combine(merr.WrappedErrors()...)
```

Comment:
Also, I'd prefer to add an API like this to our multierror package. If we copied this exactly, we'd be able to transparently support expanding nested multierrors.